File: frontend/src/models/membro.ts
export type Membro = {
    "id": string;
    "nome": string;
    "email": string;
    "cpf": string;
    "identidade": string;
    "sexo": string;
    "dataDeNascimento": string;
    "nascionalidade": string;
    "idade" : number;
    "telefone": string;
    "endereco": {
        "cep": string;
        "logradouro": string;
        "complemento": string;
        "bairro": string;
        "localidade": string;
        "uf": string;
        "ibge": string;
        "gia": string;
        "ddd": string;
        "siafi": string;
    },
    "numero": string;
    "complemento": string;
    "nomeDoPai": string;
    "nomeDaMae": string;
    "profissao": string;
    "estadoCivil": string;
    "conjuge": {
        "idMembro": string;
        "nome": string;
        "email": string;
        "cpf": string;
        "dataDeNascimento": string;
        "nascionalidade": string;
        "profissao": string;
        "ehMembro": boolean;
    },
    "dataDoCasamento": string;
    "possuiFilhos": string;
    "portadorDeNecessidadesEspeciais": string;
    "detalhamentoPne": string;
    "filhos": [
        {
            "id": string;
            "nome": string;
            "cpf": string;
            "identidade": string;
            "sexo": string;
            "dataDeNascimento": Date;
            "nascionalidade": string;
            "endereco": string;
            "numero": string;
            "pai": {
                "idMembro": string;
                "nome": string;
            },
            "mae": {
                "idMembro": string;
                "nome": string;
            },
            "portadorDeNecessidadesEspeciais": string;
            "detalhamentoPne": string;
        }
    ];
    "dataCadastro": "2023-05-14T18:21:12.189+00:00";
    "dataDaInscricao": "2023-05-14T18:21:12.189+00:00";
    "batizado": string;
    "igrejaAnterior": string;
    "igreja": {
        "id": string;
        "nome": string;
    },
    "tipoMembro": string;
    "funcoes": [
        {
            "id": "64b86e1591204b537ec4ea36",
            "descricao": "VICE LÍDER MINISTERIAL",
            "tipoFuncao": "REMUNERADO",
            "igreja": {
                "id": "64b2b8e3af06d53e804e33b9",
                "nome": "Igreja Batista Jardim Maracanã"
            },
            "estrutura": {
                "id": "64b751fa7bbb4126bc53fc57",
                "nome": "ALFA",
                "tipoGrupo": "PGM"
            }
        }
    ],
    "usuario": null
    "grupos": [
        {
            "id": string;
            "nome": string;
            "tipoGrupo": string;
        }
    ]

}

File: backend/src/main/java/com/ibjm/integraigreja/domain/Membro.java
package com.ibjm.integraigreja.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.ibjm.integraigreja.domain.dto.ConjugeDTO;
import com.ibjm.integraigreja.domain.dto.GrupoDTO;
import com.ibjm.integraigreja.domain.dto.IgrejaDTO;
import com.ibjm.integraigreja.domain.enums.EstadoCivil;
import com.ibjm.integraigreja.domain.enums.TipoMembro;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.time.temporal.ChronoField;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document
public class Membro {
    private static final long serialVersionUID = 1L;

    @Id
    private String id;
    private String nome;
    private String email;
    private String cpf;
    private String identidade;
    private Character sexo;
    private LocalDate dataDeNascimento;
    private String nascionalidade;
    private Endereco endereco;
    private String numero;
    private String complemento;
    private String nomeDoPai;
    private String nomeDaMae;
    private String profissao;
    private EstadoCivil estadoCivil;
    private ConjugeDTO conjuge;
    private Date dataDoCasamento; // remover este atributo, vai ficar no conjuge
    private String possuiFilhos;
    private String portadorDeNecessidadesEspeciais;
    private String detalhamentoPne;
    @DBRef(lazy = true)
    private List<Filho> filhos = new ArrayList<>();
    private LocalDate dataCadastro;
    private String batizado;
    private String igrejaAnterior;
    private IgrejaDTO igreja;
    private TipoMembro tipoMembro;
    @DBRef
    private List<Funcao> funcoes = new ArrayList<>();
    private Usuario usuario;
    @Transient
    private Integer idade;
    private List<GrupoDTO> grupos = new ArrayList<>();
    private String telefone;

    public Integer getIdade() {
        LocalDate now = LocalDate.now();
        long l = (now.getLong(ChronoField.EPOCH_DAY) - dataDeNascimento.getLong(ChronoField.EPOCH_DAY))/365;
        Integer idade = (Integer.parseInt(String.valueOf(l)));
        return idade;
    }
}
